### ISSUE TITLE
Content security policy (CSP)

### ISSUE THREAD

**Issue by dmitryvatagin** [verbb/formie#2050]
### What are you trying to do?

I try to add nonce parameter to the Formie script, especially to the one that contains
window.FormieTranslations

### What's your proposed solution?

add possibility to add parameters to the scripts. Not sure do I have any control over that script, beside switching off javascript in template.

### Additional context

Is there solution, how I can add nonce parameters to Formei scripts with current functionality?

**Comment by engram-design** [comment]
I've added the ability to use the following:

```twig
{{ craft.formie.renderForm(form, {
    scriptAttributes: {
        nonce: 'test',
    },
}) }}
```

This will add any attributes you define in `scriptAttributes` to every `<script>` tag Formie uses. If you want to target just the `formie.js` file, you can add:

```twig
{{ craft.formie.renderForm(form, {
    scriptAttributes: {
        nonce: 'test',
    },
    jsAttributes: {
        initJs: false,
    },
}) }}
```

These are also supported in the standalone tags:

```twig
{{ craft.formie.renderFormJs('contactForm', {
    scriptAttributes: {
        nonce: 'test',
    },
}) }}

{{ craft.formie.renderJs(false, {
    scriptAttributes: {
        nonce: 'test',
    },
}) }}

{% do craft.formie.registerAssets('contactForm', {
    scriptAttributes: {
        nonce: 'test',
    },
}) %}
```

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.28"`.

**Comment by dmitryvatagin** [comment]
@engram-design thank you! It is working 👍

**Comment by engram-design** [comment]
Fixed in [2.1.29](https://github.com/verbb/formie/releases/tag/2.1.29)
